x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LexerConfig.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.parser;
+
+import org.antlr.v4.runtime.CharStream;
+import org.antlr.v4.runtime.Lexer;
+
+/**
+ * Base class for hooking versioning information into the ANTLR parser.
+ */
+public abstract class LexerConfig extends Lexer {
+
+    EsqlConfig config = new EsqlConfig();
```

Comment:
No :)
I started with a default object (in the lexer) that could be shared by the parser but since we already configure the EsqlParser it's not needed.
Really the style depends on how Kibana uses it - I'd like the two files to be as close as possible.